src/cdef_apply.rs

```diff
@@ -161,181 +163,180 @@ pub(crate) unsafe fn rav1d_cdef_brow<BD: BitDepth>(
     lflvl: *const Av1Filter,
     by_start: c_int,
     by_end: c_int,
-    sbrow_start: c_int,
+    sbrow_start: bool,
     sby: c_int,
 ) {
-    let f: *mut Rav1dFrameContext = tc.f as *mut Rav1dFrameContext;
+    let f = &*tc.f;
     let bitdepth_min_8 = match BD::BPC {
         BPC::BPC8 => 0,
-        BPC::BPC16 => (*f).cur.p.bpc - 8,
+        BPC::BPC16 => f.cur.p.bpc - 8,
     };
-    let dsp: *const Rav1dDSPContext = (*f).dsp;
+    let dsp = &*f.dsp;
```

Comment:
`f` and `dsp` are now asserted to be valid references. That seems OK, but it is an assumption.

Reply:
Yes, that should be totally fine.  They should be fully initialized long before this I'm pretty sure.

Also, nice that `f` is no longer `&mut`.